lib/logstash/inputs/beats/message_listener.rb

```diff
@@ -119,6 +122,35 @@ def flush_buffer(ctx)
       end
     end
 
+    def extract_tls_peer(hash)
+      if @input.client_authentication_metadata?
+        tls_session = ctx.channel().pipeline().get("ssl-handler").engine().getSession()
+        tls_verified = false
+
+        # throws SSLPeerUnverifiedException if unverified
+        begin
+          tls_session.getPeerCertificates()
+          tls_verified = true
+        rescue SSLPeerUnverifiedException => e
+          if input.logger.debug?
+            input.logger.warn("Failed to get peer certificates.", :exception => e.to_s)
+          end
+        end
+
+        if tls_verified
+          hash['@metadata']['tls_peer'] = {
+            :status     => "verified",
+            :protocol   => tls_session.getProtocol(),
+            :principal  => tls_session.getPeerPrincipal().getName()
```

Comment:
Is this the CN of the peer cert ?

Reply:
Not only `CN`. It should be the complete subject `DN`, that also includes a `CN`. This is an arbitrary that client may inform to identify itself